site/content/features/sealedtypes.md

```diff
@@ -0,0 +1,506 @@
+
+As an example, consider XML node types: elements, text, comments, CDATA sections, entity references, and processing instructions.
+
+```
+sealed class Node permits Element, Text, Comment,
+      CDATASection, EntityReference, ProcessingInstruction {
+   . . .
+}
+```
+
+We might want to allow arbitrary subclasses of `Element` (as does with  `org.w3c.dom.Element`, which has dozens of HTML element subclasses). Then the declaration goes like this: 
+
+```
+non-sealed class Element extends Node {
+   . . .
+}
+```
+
+# New Keywords and Restricted Identifiers
+
+The tokens `sealed` and `permits` are *restricted identifiers* that have a special meaning only in class and interface declarations, just like `record`, `var`, and `yield`. Code with variables named `sealed` and `permits` won't break. But you can no longer define classes named `sealed` and `permits`:
+
+```
+sealed class sealed permits permits {}
+final class permits extends sealed {}
+```
+
+```
+public class Sandbox {
+   public static void main(String[] args) {
+      System.out.println("Move along, nothing to see here...");
+   }
+}
+```
+
+In contrast, `non-sealed` is a keyword. Obviously, you cannot use it as an identifier since it contains a `-` character. In fact, `non-sealed` is the second keyword that contains a character that isn't a lowercase letter. The first one is `_`, since Java 9. I am sure
+
+And yes, you can continue to compute the difference of two variables `non` and `sealed`:
+
+```
+public class Sandbox {
+   public static void main(String[] args) {
+      int non = 7 * 7;
+      int sealed = 7;
+      System.out.println(non-sealed);
+   }
+}
+```
+
+## Package/Module Restriction
+
+If you don't use modules, then the sealed class and its direct subclasses must be in the same package. If you use modules, they must all be in the same module.
+
+After all, these classes are developed and maintained together, so there should be no reason to spread them far and wide. 
+
+There is one vexing situation. If you don't want to use modules, you cannot put the superclass into an API package and the subclasses into a separate implementation package.  
+
+## Omitting the Permits Clause
+
+If the subclasses of a sealed class are all defined in the same source file, then you can omit the `permits` clause: 
+
+```
+import java.io.*;
+import java.nio.file.*;
+import java.util.function.*;
+import java.util.stream.*;
+
+sealed class DirectoryEntry { // No permits clause
+   private Path path;
+   public Path getPath() { return path; }
+
+   // Direct subclasses in same source file
+   public static final class Directory extends DirectoryEntry {
+      private Directory() {}
+      public Stream<DirectoryEntry> entries() {
+         try {
+            return Files.list(getPath()).map(DirectoryEntry::of);
+         } catch (AccessDeniedException ex) {
```

Comment:
What about simply declaring IOException to make the example more concise?

Reply:
Ah, doesn't work as you can't handle the checked exception in the `Consumer` API.